src/accelerate/utils/imports.py

```diff
@@ -45,7 +52,18 @@ def is_tpu_available():
 
 
 def is_deepspeed_available():
-    return importlib.util.find_spec("deepspeed") is not None
+    package_exists = importlib.util.find_spec("deepspeed") is not None
+    # Check we're not importing a "deepspeed" directory somewhere but the actual library by trying to grab the version
+    # AND checking it has an author field in the metadata that is HuggingFace.
+    if package_exists:
+        try:
+            _ = importlib_metadata.version("deepspeed")
```

Comment:
```suggestion
```
If I'm not mistaken this is unnecessary. You can do `_deepspeed_metadata.get("version")` if it's needed?

Reply:
`importlib_metadata.metadata` will also then raise the same error if it's not installed